src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -83,13 +83,18 @@ public static class TagHelperDescriptorFactory
             bool designTime)
         {
             TagHelperUsageDescriptor typeUsageDescriptor = null;
+            string outputElementHint = null;
 
-#if !DNXCORE50
             if (designTime)
             {
+                var outputElementHintAttribute =
+                    typeInfo.GetCustomAttribute<OutputElementHintAttribute>(inherit: false);
+
+                outputElementHint = outputElementHintAttribute?.OutputElement;
```

Comment:
Doubt the `GetCustomAttribute<T>()` is so expensive that it needs to be inside this condition. Also this factory should only execute once at runtime. So calculate the hint unconditionally.